pkg/common/utils.go

```diff
@@ -17,3 +19,13 @@ func RemoveStringSliceItem(item string, slice *[]string) {
 		}
 	}
 }
+
+// UnmarshalJSON is a helper function to JSON unmarshal an encoded object into
+// a concrete type T.
+func UnmarshalJSON[T any](data []byte) (*T, error) {
```

Comment:
Do we have any safeguards against someone trying to use this to unmarshal into a protobuf (which you're not supposed to use encoding/json for?)

Reply:
No, hopefully the name of the function is enough of a deterrent, though. It's currently not being used (and may not have to be used), so I'll remove it from this PR.